lib/matplotlib/cm.py

```diff
@@ -22,10 +24,11 @@
 # reversed colormaps have '_r' appended to the name.
 
 
-def _reverser(f):
-    def freversed(x):
-        return f(1 - x)
-    return freversed
+def _reverser(f, x=None):
+    if x is None:
```

Comment:
This could use a short explanation even if it's private. Something like


~~~
Return a function f' for a given function f such that f'(x) = f(1-x).

Usage: ``_reverser(f)``. The parameter *x* is only used internally.
~~~

Maybe also use `_x` instead of `x`.